### DISCUSSION TITLE
Customizing the scroll bar of the TextEditor control

### DISCUSSION THREAD

**Discussion by kokaKOLA-NEW** [AvaloniaUI/AvaloniaEdit#416] [Q&A] [answered]
Hello,

I am trying to customize the scroll bar of the TextEditor control in my Avalonia application. I know how to style scroll bars in general, but I am having difficulty applying these styles to the TextEditor's scroll bar.

Does anyone know how I can customize the scroll bar of the TextEditor please?

**Comment by mgarstenauer** [comment, marked as answer]
Here is an example: I added a style to *AvaloniaEdit.Demo* to make the vertical scrollbar background red:

```xaml
<Window xmlns="https://github.com/avaloniaui"
        xmlns:AvalonEdit="using:AvaloniaEdit"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        x:Class="AvaloniaEdit.Demo.MainWindow" ...>

  <Window.Styles>
    <Style Selector="AvalonEdit|TextEditor.red /template/ ScrollViewer /template/ ScrollBar#PART_VerticalScrollBar">
      <Setter Property="Background" Value="Red"/>
    </Style>
  </Window.Styles>

    ...

        <AvalonEdit:TextEditor Classes="red"
                               HorizontalScrollBarVisibility="Auto"
                               VerticalScrollBarVisibility="Visible" />

    ...
</Window>
```

**Comment by kokaKOLA-NEW** [comment]
Thank you very much

**Comment by kokaKOLA-NEW** [comment]
Sorry to bother you again, but do you know how I can make the content of the text have a right margin? Because the scrollbar is currently overlapping my scroll bar.

**Comment by mgarstenauer** [comment]
Try: 

```xaml
<AvalonEdit:TextEditor Padding="0,0,50,0" ...  />
```

**Reply by kokaKOLA-NEW** [reply]
I hadn't thought of that, thank you very much.
